### DISCUSSION TITLE
"command not found: vercel" after installing globally

### DISCUSSION THREAD

**Discussion by lukosevicius** [vercel/vercel#5019] [Help] [answered]
I just installed vercel with:
`npm install -g vercel`
But when I try to run: `verlcel`, I get: `zsh: command not found: vercel`

I tried to list installed NPM packaged to see if it is installed, and seems that it is:

![image](https://user-images.githubusercontent.com/23518661/89795321-ea0f3600-db30-11ea-8941-ffec699e4fdb.png)

How could I make it run?

**Comment by mcsdevv** [comment]
Hey @lukosevicius, this is likely an issue with your Node.js installation, are you able to run other commands globally?

**Comment by StayFoolisj** [comment]
Hi Mantas did you ever figure this out? I'm having the same issue

**Reply by lukosevicius** [reply]
Hi, no, I didn't. I moved on to alternative.

**Reply by salatielosorno** [reply]
I have the same problem. What was the alternative? Could you share it?

**Reply by solarlime** [reply]
The same. I installed it via yarn, so it's located in ~/.yarn/bin/. Fixing the PATH should help:
`export PATH=$PATH:~/.yarn/bin`
The same solution for npm, just change the route.

**Reply by Blueblack319** [reply]
where to fix path??

**Reply by TheKachi** [reply]
@solarlime Please can you be a little more explicit about how to fix path. I need the exact command to run

**Reply by francesca-giammichele-mahana** [reply]
@TheKachi , i just typed the command in the terminal and it worked
export PATH=$PATH:~/.yarn/bin

**Comment by TomVizzo** [comment, marked as answer]
Solution:
`mkdir ~/.npm-global`
`export NPM_CONFIG_PREFIX=~/.npm-global`
`export PATH=$PATH:~/.npm-global/bin`
`echo -e "export NPM_CONFIG_PREFIX=~/.npm-global\nexport PATH=$PATH:~/.npm-global/bin" >> ~/.bashrc`

**Reply by Joshua-Ogbonna** [reply]
This helped! Thank you very much

**Reply by codewithjaylab** [reply]
This helped! Thank your very much X2

**Reply by codeMagic122** [reply]
This helped me so much! I really appreciate the straightforward solution!

To other noobs who come across this thread, just copy paste the solution! PATH and NPM_CONFIG_PREFIX aren't variable names that you have to change :D

**Reply by brenocastelo** [reply]
This helped me. Thanks!

**Reply by eulixir** [reply]
Worked!, Thank you :)

**Reply by guadalupe182** [reply]
esto funciono! muchas gracias !!

**Reply by Leoactionz** [reply]
> Solution: `mkdir ~/.npm-global` `export NPM_CONFIG_PREFIX=~/.npm-global` `export PATH=$PATH:~/.npm-global/bin` `echo -e "export NPM_CONFIG_PREFIX=~/.npm-global\nexport PATH=$PATH:~/.npm-global/bin" >> ~/.bashrc`

Very helpful ty so much

**Reply by Ashkur** [reply]
I have done it, and now i have an error every time i open bash `bash: /c/Users/Uriel/.bashrc: line 2: syntax error near unexpected token `('`.

Also, didn't solve the problem either. Can somebody help, please?  😢

**Reply by shockz09** [reply]
tysm

**Reply by maddie-wang** [reply]
worked

**Reply by BrendonLH** [reply]
My classmate was having this issue and these commands fixed it!! Thank you so much!

**Reply by brianjtarte** [reply]
Lifesaver!

**Reply by faizabaig13** [reply]
Thank It's Working

**Reply by mercpls** [reply]
Any solution for using yarn?

**Reply by Amar-arruf** [reply]
it's not worked on Windows. so how I can solved it ?

**Reply by paulamurta** [reply]
thank you so much

**Reply by maninkari** [reply]
@mercpls for yarn I did `export PATH=$PATH:~/.yarn/bin`

**Reply by rkrakib237** [reply]
any permanent solution ? everytime need to use this command before work

**Reply by amyegan** [reply]
If you have added it to your `zsh` or `bash` config, you shouldn't need to manually run it every time you restart. Examples: https://github.com/vercel/vercel/discussions/5019#discussioncomment-333260 and https://github.com/vercel/vercel/discussions/5019#discussioncomment-3563475

**Reply by satyamasai** [reply]
not helpful

**Reply by mdfaysal22** [reply]
this is not working parentally

**Reply by JKQuah** [reply]
Not working

**Reply by Sir-Rumeh** [reply]
Where exactly am i supposed to trigger these commands please

**Reply by abrahammegantoro** [reply]
can you provide your code for powershell, please? Thank you in advance :)

**Reply by truecode112** [reply]
It worked for me.
Thanks

**Reply by Johnrobmiller** [reply]
please be sure to mention what OS this is running on when pasting command lines like this. to not do so is to fail to put yourself in another person's shoes -- not everybody uses the same OS your are.

**Reply by kevalRabadiya** [reply]
not working

**Comment by kkdima** [comment]
@lukosevicius @mcsdevv @salatielosorno @StayFoolisj @solarlime 
Hey guys! I encountered the same problem tried approaches you mentioned here and still can't resolve it.
I found where I have `vercel` installed with the command: `npm list -g --depth=0` and specifically added it to my $PATH settings.

**Command `printenv` show me this:**
```PATH=/Users/dimakononenko/.npm-packages/lib:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:/usr/X11/bin:/Users/dimakononenko/bin ```

But calling "vercel" in the CLI still gives this error `zsh: command not found: vercel
`
Did anyone come close to understanding the nature of this issue?

**Comment by vindecodex** [comment]
@lukosevicius

It's just that you've been using `zsh` instead of `bash`, you can add your npm bin PATH to your `zsh` config.

`npm get prefix` - returns the path of your node that contains the bin directory.

```
echo "export PATH=$PATH:$(npm get prefix)/bin" >> ~/.zshrc"
```

`source ~/.zshrc` - refresh your config to take effect

You can even double check if vercel exist inside the bin folder, this is where global packages can be found.
If your switching to `bash` better to add it also on your `~/.bashrc'

**Reply by duggiemitchell** [reply]
This worked for me when using npm and zsh 💅🏾, but small typo:

`echo "export PATH=$PATH:$(npm get prefix)/bin" >> ~/.zshrc`

**Reply by proverbed** [reply]
This worked for me, thank you 🙏

**Comment by rimzzlabs** [comment]
hello is there any solution to this? Im using yarn but still doesn't work

**Reply by romelgomez** [reply]
try with this. 

- `echo "export PATH=$PATH:$(yarn global bin)" >> ~/.zshrc`
- `source ~/.zshrc`

**Reply by leerob** [reply]
Yeah, or potentially your `.bash_profile` if you're not using ZSH!

**Reply by naskritus** [reply]
Thank you!

**Comment by ghost** [comment]
I use node with the asdf version manager. `asdf reshim` solved this issue for me

**Comment by mobagenie** [comment]
if command 

`export PATH=$PATH:~/.yarn/bin`

or 

`export NPM_CONFIG_PREFIX=~/.npm-global`

still not work, you can force run with full path to binary vercel :

```
mg@mg-PC:~/.yarn/bin/vercel --version
Vercel CLI 28.12.4
28.12.4
```

**Reply by jauresmanouan** [reply]
`export PATH=$PATH:~/.yarn/bin` work perfectly thanks 👍🏾

**Reply by lokinmodar** [reply]
this worked for me

**Comment by NatumanyaGuy** [comment]
This happened to me when I deleted my Path user environment variables (on Windows!).

- Go to the npm globally installed packages here - `C:\Users\<username>\AppData\Roaming\npm`
- Confirm that vercel exists - as a couple of files (e.g. _vercel, vercel.cmd, vercel.ps1_)
- Copy this directory value (`C:\Users\<username>\AppData\Roaming\npm`)
- Add this directory value to your Path **_user environment variables_**.
- Restart your computer and then run `vercel -v` in the CLI (could be _Powershell_ or _Command Prompt_) to confirm.

**Reply by Ayush242** [reply]
I am able to run vercel in cmd but not on bash.
P.S. above solutions didn't worked for me

**Reply by rd273001** [reply]
It worked on CMD outside VS code, but not working on vsCode terminal either using bash/CMD/PowerShell?

**Comment by kingsleynwoye** [comment]
I successfully installed Vercel within macOS using the following command.
`brew install vercel-cli`

**Reply by Risini1998** [reply]
this works with npm also

**Reply by arnelamo** [reply]
This works

**Reply by gillianroberts1** [reply]
this works!

**Reply by ivonamojsovska** [reply]
amin

**Reply by jornatf** [reply]
Thank you @kingsleynwoye this works for me!! You saved my life.

**Comment by brandonleetran** [comment]
I am using yarn.. anybody found an actual fix for this

**Reply by prodkt** [reply]
Nothing. We have the feeling we've exhausted this as much as we can without an official response from vercel.

**Reply by gonwi** [reply]
Just add the following `export PATH=$PATH:~/.yarn/bin` at the end of your `.bashrc` or `zshrc` and then `source .bashrc`.

**Comment by jonthornham** [comment]
Running this command in terminal worked for me. I used npm to install Vercel. 

`alias vercel="`npm config get prefix`/bin/vercel"`
